File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    scores = dict()
    for i in range(len(name)):
        scores[name[i]] = yearning[i]
    
    for ph in photo:
        total = 0
        for nm in ph:
            if nm not in scores.keys():
                continue
            else:
                total += scores[nm]
        answer.append(total)
        
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int n = photo.length; // 사진 개수
        int[] answer = new int[n];
        
        Map<String, Integer> scores = new HashMap<>();
        for (int i = 0; i < name.length; i++) {
            scores.putIfAbsent(name[i], yearning[i]);
        }
        
        for (int i = 0; i < n; i++) {
            int total = 0;
            for (String nm: photo[i]) {
                if (scores.containsKey(nm)) total += scores.get(nm);
            }
            answer[i] = total;
        }
        
        return answer;
    }
}